packages/eas-cli/src/commands/build/index.ts

```diff
@@ -140,17 +142,29 @@ async function promptForPlatformAsync(): Promise<RequestedPlatform> {
   return platform;
 }
 
-async function ensureProjectConfiguredAsync(projectDir: string): Promise<ExpoConfig | null> {
-  if (await fs.pathExists(path.join(projectDir, 'eas.json'))) {
+async function ensureProjectConfiguredAsync(
+  projectDir: string,
+  platform: RequestedPlatform
+): Promise<ExpoConfig | null> {
+  const platformsToConfigure = await getPlatformsToConfigureAsync(projectDir, platform);
+
+  if (!platformsToConfigure) {
     return null;
   }
-  const confirm = await confirmAsync({
-    message: 'This app is not set up for building with EAS. Set it up now?',
-  });
+
+  // Ensure the prompt is consistent with the platforms we need to configure
+  let message = 'This app is not configured to build with EAS. Set it up now?';
+  if (platformsToConfigure === RequestedPlatform.Ios) {
+    message = 'Your iOS app is not configured to build with EAS. Set it up now?';
+  } else if (platformsToConfigure === RequestedPlatform.Android) {
+    message = 'Your Android app is not configured to build with EAS. Set it up now?';
+  }
```

Comment:
Shouldn't we change `app` -> `project`? E.g. `Your Android project is not configured to build with EAS. Set it up now?`